pkg/proxy/proxy.go

```diff
@@ -96,16 +97,26 @@ type Proxy struct {
 	// Datapath updater for installing and removing proxy rules for a single
 	// proxy port
 	datapathUpdater DatapathUpdater
+
+	// IPCache is used for tracking IP->Identity mappings and propagating
+	// them to the proxy via NPHDS in the cases described
+	ipcache *ipcache.IPCache
+
+	// defaultEndpointInfoRegistry is the default instance implementing the
+	// EndpointInfoRegistry interface.
+	defaultEndpointInfoRegistry *endpointInfoRegistry
```

Comment:
I left the default part in there since it seemed like @jrajahalme had some specific intention around that. Probably related to it being only used for the DNS proxy but not for Envoy. :shrug: